gp_align/analysis.py

```diff
@@ -94,21 +102,29 @@ def analyze_run(images, scanner=1, plate_type=1, orientation="top-right",
 
     output = dict()
     columns = config["well_names"]
-    columns.insert(0, "time")
+    index = config["index_name"]
+    columns.insert(0, index)
     for plate, plate_data in iteritems(data):
         plate_df = DataFrame(plate_data)
         assert len(plate_df.columns) == len(columns), "{:d} != {:d}".format(
             len(plate_df.columns), len(columns))
-        plate_df.sort_values("time", inplace=True)
-        plate_df["time"] -= plate_df["time"].iat[0]
-        plate_df["time"] /= unit
-        plate_df.set_index("time", inplace=True)
-        output[plate] = plate_df.loc[:, columns[1:]]  # order columns
+        plate_df.sort_values(index, inplace=True)
+        if parse_timestamps:
+            plate_df[index] -= plate_df[index].iat[0]
+            plate_df[index] /= unit
+        plate_df.set_index(index, inplace=True)
+        ordered_wells = well_names(config["rows"], config["columns"], orientation="top-left")
+        output[plate] = plate_df[ordered_wells]  # order columns
```

Comment:
Do you always want the output order of 'top-left'? Otherwise I would say just use `columns[1:]` as before but without the `.loc` as you do here. Should be fine.

Reply:
Yes the output should be in the conventional A1-H12 order